lib/AST/ASTImporter.cpp

```diff
@@ -5517,10 +5552,32 @@ ASTNodeImporter::VisitFunctionTemplateDecl(FunctionTemplateDecl *D) {
           }
           FoundByLookup = FoundTemplate;
           break;
+        } else {
```

Comment:
However, we should check for structural eq in case of function template **sepcializations** (they do not overload).
See also https://github.com/Ericsson/clang/issues/585

Reply:
Nice catch, thanks for the insight! Will update the PR to include the relevant comments.